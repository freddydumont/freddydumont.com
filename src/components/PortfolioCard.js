/** @jsx jsx **/
import { jsx } from 'theme-ui';
import { useState, Fragment } from 'react';
import Img from 'gatsby-image';
import { SlideDown } from 'react-slidedown';
import { Text, Box, Button, Flex, Card, Badge } from '@theme-ui/components';
import ButtonLink from './ButtonLink';

/**
 * Tag object with name and ThemeUI color
 * @typedef {Object} Tag
 * @property {String} name
 * @property {String} color text color
 * @property {String} bg background color
 */

/**
 * Turn portfolio data into a card component.
 * @param {Object} props
 * @param {String} props.title
 * @param {String} props.category
 * @param {String} props.description
 * @param {String} props.technology
 * @param {String} props.link
 * @param {String} props.alt
 * @param {Object} props.image
 * @param {Tag[]} props.tags
 */
const PortfolioCard = ({
  title,
  category,
  description,
  technology,
  link,
  alt,
  image,
  tags,
}) => {
  const [isExpanded, setExpanded] = useState(false);

  return (
    <Card
      bg="gray.8"
      sx={{
        borderTopWidth: 4,
        borderTopColor: 'teal_vivid.6',
        position: 'relative',
      }}
    >
      <Box px={6} pt={4} pb={3}>
        <Text as="h2" variant="title">
          {title}
        </Text>
      </Box>
      <Img
        fluid={image.childImageSharp.fluid}
        alt={alt}
        sx={{
          marginBottom: 4,
          marginX: 2,
          borderRadius: 'sm',
        }}
      />
      <SlideDown
        transitionOnAppear={false}
        sx={{
          transition: `transform 0.3s cubic-bezier(0.23, 1, 0.32, 1)`,
        }}
      >
        <Box px={6} py={4}>
          <Text as="h3" mb={4} variant="category">
            {category}
          </Text>
          <Flex
            data-testid="card-tag-container"
            mb={2}
            css={{
              flexWrap: 'nowrap',
              overflowX: 'scroll',
              WebkitOverflowScrolling: 'touch',
            }}
          >
            {tags.map((tag) => (
              <Tag key={tag.name} {...tag} />
            ))}
          </Flex>
          <Text as="p" variant="description">
            {description}
          </Text>
          {isExpanded && (
            <Fragment>
              <Text
                as="h3"
                mt={4}
                mb={1}
                color="gray.1"
                sx={{ fontSize: 'lg', fontFamily: 'body' }}
              >
                Technology
              </Text>
              <Text as="p" variant="description">
                {technology}
              </Text>
            </Fragment>
          )}
        </Box>
        {isExpanded ? (
          <ButtonLink link={link}>go to project</ButtonLink>
        ) : (
          <Button
            data-testid="expand-button"
            sx={{
              position: 'absolute',
              right: 0,
              left: 0,
              bottom: 0,
              filter: `drop-shadow(0 0 20px)`,
            }}
            onClick={() => setExpanded(true)}
          >
            See More
          </Button>
        )}
      </SlideDown>
    </Card>
  );
};

export const Tag = ({ name, color, bg, ...props }) => (
  <Badge as="span" color={color} bg={bg} variant="tech_tag" {...props}>
    {name}
  </Badge>
);

export default PortfolioCard;
