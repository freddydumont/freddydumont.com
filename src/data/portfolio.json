[
  {
    "title": "NessIA.ca",
    "category": "Business website",
    "description": "Built for a client operating in the Business Intelligence space, this website outperforms competitors in speed, design and usability. It serves as a showcase for their services and as social proof via the Client and Partners section.",
    "technology": "The tool of choice these days for blazing fast websites is Gatsby. Its plugin architecture and Node API make website development a breeze.",
    "link": "https://nessia.ca/en",
    "image": "./src/assets/nessia.png",
    "alt": "NessIA homepage",
    "tags": ["javascript", "react", "gatsby", "bulma"]
  },
  {
    "title": "Huddol",
    "category": "Front End Development",
    "description": "React front end development work on the Huddol website and platform as part of a small development team. I mentored junior developers and cooperated with the CTO and my peers to get ideas to production as fast.",
    "technology": "Typical React/Redux application. Most of the codebase was of the early React adoption days so a major challenge which I enjoyed was to refactor enormous class components into well designed functional hook-based components for easier maintainability.",
    "link": "https://huddol.com",
    "image": "./src/assets/huddol_homepage.png",
    "alt": "Huddol homepage",
    "tags": ["javascript", "react", "redux", "jest", "emotion"]
  },
  {
    "title": "Digispeed.io",
    "category": "Business Website",
    "description": "Built for a client who needed an online presence for their business. The provided company logo influenced the design decisions. We chose to keep a simple document feeling to complement its black and white nature. Smooth animations and well designed icons improve the usability of the site.",
    "technology": "I believe in using the right tool for the job at hand. In this case, we needed a simple framework to quickly design and build a landing page. Gatsby is the obvious choice in this category, having all the features needed for navigation and extensibility. Bootstrap and Sass are used to power the CSS, and Netlify deploys the website at minimal costs.",
    "link": "https://digispeed.io",
    "image": "./src/assets/digispeed.png",
    "alt": "Digispeed homepage",
    "tags": ["javascript", "react", "gatsby", "bootstrap", "sass"]
  },
  {
    "title": "Netsweeper",
    "category": "Game",
    "description": "Neon cyberpunk minesweeper. Built for fun and to learn Phaser and Typescript.",
    "technology": "Netsweeper is built in TypeScript using the Phaser game engine. Since DOM Element Game Objects weren't supported in Phaser 3.16, React was used to create UI elements positioned on Game Objects. An xstate State Machine handles most of the gameplay logic.",
    "link": "https://freddydumont.itch.io/netsweeper",
    "image": "./src/assets/netsweeper.png",
    "alt": "netsweeper gameplay",
    "tags": ["typescript", "phaser", "react", "xstate", "emotion"]
  },
  {
    "title": "Star Wars Planet Guide",
    "category": "Typical interview project",
    "description": "A tool for Star Wars fans to browse and learn about the different planets of the Star Wars extended universe.",
    "technology": "The Planet Guide makes use of redux-saga to handle asynchronous events such as the API calls to SWAPI. The DevExtreme React Grid component is used to present the data in a table handling filtering, sorting and the display of detailed information.",
    "link": "https://freddydumont.github.io/star-wars-planet-guide/",
    "image": "./src/assets/starwars.png",
    "alt": "star wars planet guide",
    "tags": ["javascript", "react", "redux", "redux-saga", "bootstrap"]
  }
]
