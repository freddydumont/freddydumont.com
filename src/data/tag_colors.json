{
  "javascript": "bg-yellow-vivid-400 text-gray-900",
  "react": "bg-blue-400 text-gray-900",
  "gatsby": "bg-purple-700 text-purple-100",
  "bulma": "bg-teal-vivid-200 text-teal-vivid-900",
  "redux": "bg-purple-600 text-purple-100",
  "jest": "bg-red-vivid-800 text-white",
  "emotion": "bg-pink-400 text-gray-900"
}
