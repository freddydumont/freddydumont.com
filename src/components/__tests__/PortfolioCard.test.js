import React from 'react';
import { render, fireEvent, wait } from '@testing-library/react';
import PortfolioCard from '../PortfolioCard';
import data from './fixtures/card_data.json';

describe('PortfolioCard', () => {
  const { queryByText, queryByTestId } = render(<PortfolioCard {...data} />);
  const technology = queryByText(/technology/i);
  const expandButton = queryByTestId('expand-button');

  it('should hide the technology section before expanding', () => {
    expect(technology).not.toBeInTheDocument();
    expect(queryByText(/go to project/i)).not.toBeInTheDocument();
  });

  it('should have an expand button', () => {
    expect(expandButton).toBeVisible();
  });

  it('should expand when clicking the expand button', async () => {
    const { queryByText, queryByTestId } = render(<PortfolioCard {...data} />);

    const button = queryByTestId('expand-button');

    fireEvent.click(button);

    await wait(() => {
      expect(queryByText(/technology/i)).toBeVisible();
      expect(button).not.toBeInTheDocument();
      expect(queryByText(/go to project/i)).toBeVisible();
    });
  });
});
