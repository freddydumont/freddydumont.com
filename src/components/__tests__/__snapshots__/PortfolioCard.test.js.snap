// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioCard should match snapshot 1`] = `
.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  min-width: 20rem;
  max-width: 20rem;
  border-radius: 0.25rem;
  overflow: hidden;
  box-shadow: 0 10px 15px -3px rgba(0,0,0,0.1),0 4px 6px -2px rgba(0,0,0,0.05);
  margin-bottom: 2rem;
  margin-left: auto;
  margin-right: auto;
  background-color: var(--theme-ui-colors-gray-8,#3B3B3B);
  border-top-width: 4px;
  border-top-color: var(--theme-ui-colors-teal_vivid-6,#079A82);
  position: relative;
}

@media screen and (min-width:640px) {
  .emotion-13 {
    max-width: 24rem;
  }
}

@media screen and (min-width:768px) {
  .emotion-13 {
    max-width: 20rem;
  }
}

@media screen and (min-width:1024px) {
  .emotion-13 {
    max-width: 24rem;
  }
}

.emotion-13:hover {
  box-shadow: 0 25px 50px -12px rgba(0,0,0,0.25);
}

@media (max-width:360px) {
  .emotion-13 {
    min-width: 18rem;
    max-width: 18rem;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  padding-top: 1rem;
  padding-bottom: 0.75rem;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: "IBM Plex Sans",sans-serif;
  font-weight: 600;
  color: var(--theme-ui-colors-gray-0,#F7F7F7);
  font-size: 1.5rem;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
}

.emotion-2 {
  margin-bottom: 1rem;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
  border-radius: 0.125rem;
}

.emotion-12 {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.23,1,0.32,1);
  -webkit-transition: transform 0.3s cubic-bezier(0.23,1,0.32,1);
  transition: transform 0.3s cubic-bezier(0.23,1,0.32,1);
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 1.25rem;
  color: var(--theme-ui-colors-gray-1,#E1E1E1);
  margin-bottom: 1rem;
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  overflow-x: scroll;
  -webkit-overflow-scrolling: touch;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: inline-block;
  vertical-align: baseline;
  font-size: 0;
  font-weight: 700;
  white-space: nowrap;
  padding-left: 0.25rem;
  padding-right: 0.25rem;
  border-radius: 2px;
  color: var(--theme-ui-colors-white,#fff);
  background-color: var(--theme-ui-colors-primary,#17B897);
  min-width: initial;
  display: inline-block;
  font-family: "IBM Plex Sans",sans-serif;
  font-weight: 600;
  font-size: 0.875rem;
  border-radius: 0.5rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  margin-right: 0.5rem;
  margin-bottom: 0.5rem;
  white-space: nowrap;
  color: bg-yellow-vivid-400;
}

@media screen and (min-width:640px) {
  .emotion-4 {
    color: text-gray-900;
  }
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: inline-block;
  vertical-align: baseline;
  font-size: 0;
  font-weight: 700;
  white-space: nowrap;
  padding-left: 0.25rem;
  padding-right: 0.25rem;
  border-radius: 2px;
  color: var(--theme-ui-colors-white,#fff);
  background-color: var(--theme-ui-colors-primary,#17B897);
  min-width: initial;
  display: inline-block;
  font-family: "IBM Plex Sans",sans-serif;
  font-weight: 600;
  font-size: 0.875rem;
  border-radius: 0.5rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  margin-right: 0.5rem;
  margin-bottom: 0.5rem;
  white-space: nowrap;
  color: bg-blue-400;
}

@media screen and (min-width:640px) {
  .emotion-5 {
    color: text-gray-900;
  }
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: inline-block;
  vertical-align: baseline;
  font-size: 0;
  font-weight: 700;
  white-space: nowrap;
  padding-left: 0.25rem;
  padding-right: 0.25rem;
  border-radius: 2px;
  color: var(--theme-ui-colors-white,#fff);
  background-color: var(--theme-ui-colors-primary,#17B897);
  min-width: initial;
  display: inline-block;
  font-family: "IBM Plex Sans",sans-serif;
  font-weight: 600;
  font-size: 0.875rem;
  border-radius: 0.5rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  margin-right: 0.5rem;
  margin-bottom: 0.5rem;
  white-space: nowrap;
  color: bg-purple-700;
}

@media screen and (min-width:640px) {
  .emotion-6 {
    color: text-purple-100;
  }
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: inline-block;
  vertical-align: baseline;
  font-size: 0;
  font-weight: 700;
  white-space: nowrap;
  padding-left: 0.25rem;
  padding-right: 0.25rem;
  border-radius: 2px;
  color: var(--theme-ui-colors-white,#fff);
  background-color: var(--theme-ui-colors-primary,#17B897);
  min-width: initial;
  display: inline-block;
  font-family: "IBM Plex Sans",sans-serif;
  font-weight: 600;
  font-size: 0.875rem;
  border-radius: 0.5rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  margin-right: 0.5rem;
  margin-bottom: 0.5rem;
  white-space: nowrap;
  color: bg-teal-vivid-200;
}

@media screen and (min-width:640px) {
  .emotion-7 {
    color: text-teal-vivid-900;
  }
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: "IBM Plex Sans",sans-serif;
  color: var(--theme-ui-colors-gray-3,#B1B1B1);
  font-size: 1rem;
  text-align: justify;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  color: var(--theme-ui-colors-white,#fff);
  background-color: var(--theme-ui-colors-primary,#17B897);
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  display: inline-block;
  width: 100%;
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
  text-align: center;
  font-size: 1.125rem;
  font-weight: 700;
  font-family: "Overpass Mono",Menlo,Monaco,Consolas,"Liberation Mono","Courier New",monospace;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  -webkit-transition: all 100ms ease-in-out;
  transition: all 100ms ease-in-out;
  border-radius: inherit;
  background-color: #016457;
  color: #5FE3C0;
  position: absolute;
  right: 0;
  left: 0;
  bottom: 0;
  -webkit-filter: drop-shadow(0 0 20px);
  filter: drop-shadow(0 0 20px);
}

.emotion-11:hover {
  background-color: #2DCCA7;
  color: #004440;
}

<div>
  <div
    class="emotion-13"
  >
    <div
      class="emotion-1"
    >
      <h2
        class="emotion-0"
      >
        NessIA.ca
      </h2>
    </div>
    <div
      class="emotion-2 gatsby-image-wrapper"
      style="position: relative; overflow: hidden;"
    >
      <div
        aria-hidden="true"
        style="width: 100%; padding-bottom: 57.48031496062992%;"
      />
      <img
        alt=""
        aria-hidden="true"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAALCAYAAAB/Ca1DAAAACXBIWXMAAAsSAAALEgHS3X78AAACHUlEQVQoz31TTWsTQRjOn7BCf4N/QUEFQfFSb3pVvIiXInjwA2yhNBREPYiEaqAtemibIJZa6QfFXHJQsCaEpGlCst+f2cTdZHezu3mcnTFpWtSBh33m3ZlnnnnfdxKDwQDxKGkKeF2jPI4N4+O83+/D9336dV2XcqPXRdvtjdYkhiQIQ4pxkXFEUTTiIeHRPzAS/N+IF6qaRqEbJhRVg6yqkBQFkixDlBWIkkyROO0oJC5Pgwnq0HSdCBpko4RqrQZeENghusH+kZRRQcdxUC6XUalU0Gw2IZENoiiC53lYlkUPik83TAsd20aD43FQKKHeFNFqd9Cy2hBEGZxuHws2OA71RoPyIAhOFCCex7x6WMZWdhMLKzxuJTncmX6N1eU0Hi6KuDF7iPTMChMMPB+u2UKPQCDXiF3KJDcKyVHsMIpCdB0fD57ncC/zHvObBdx+IeHc1RSSyVeY+yBi6lkVU9cXmeCRaeDR7mfM5/Zgd7vwPY+6Ynllhdn9XsTEtSzuft3BkWHh3UcRk+eXcX/6CTHAYe5tBROXvjBBwe5gqfgDq+UiwrGqs3ahdcbjtQzOXM7gys0Z5HN5bOctTF7M4unLLTiuj9RaE2cvbPy9bU72IBPcKx3gzfpPLKT2kdnYQUMysb4tIJ0t4FuhDl7+haVPteO2CYiV4E/zDmMYYDSPue06cELyQsjEcrskFpAidiArKjzPJQ+jj990GCN/jrWOHwAAAABJRU5ErkJggg=="
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
      />
      <picture>
        <source
          sizes="(max-width: 384px) 100vw, 384px"
          srcset="/static/2e38d778d3d7f1a9c244a8bede600333/32a58/nessia.png 96w,
/static/2e38d778d3d7f1a9c244a8bede600333/2c007/nessia.png 192w,
/static/2e38d778d3d7f1a9c244a8bede600333/75125/nessia.png 384w,
/static/2e38d778d3d7f1a9c244a8bede600333/b6f2d/nessia.png 576w,
/static/2e38d778d3d7f1a9c244a8bede600333/17fa3/nessia.png 768w,
/static/2e38d778d3d7f1a9c244a8bede600333/6d185/nessia.png 2794w"
        />
        <img
          alt="NessIA homepage"
          loading="lazy"
          sizes="(max-width: 384px) 100vw, 384px"
          src="/static/2e38d778d3d7f1a9c244a8bede600333/75125/nessia.png"
          srcset="/static/2e38d778d3d7f1a9c244a8bede600333/32a58/nessia.png 96w,
/static/2e38d778d3d7f1a9c244a8bede600333/2c007/nessia.png 192w,
/static/2e38d778d3d7f1a9c244a8bede600333/75125/nessia.png 384w,
/static/2e38d778d3d7f1a9c244a8bede600333/b6f2d/nessia.png 576w,
/static/2e38d778d3d7f1a9c244a8bede600333/17fa3/nessia.png 768w,
/static/2e38d778d3d7f1a9c244a8bede600333/6d185/nessia.png 2794w"
          style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
        />
      </picture>
      <noscript>
        &lt;picture&gt;&lt;source srcset="/static/2e38d778d3d7f1a9c244a8bede600333/32a58/nessia.png 96w,
/static/2e38d778d3d7f1a9c244a8bede600333/2c007/nessia.png 192w,
/static/2e38d778d3d7f1a9c244a8bede600333/75125/nessia.png 384w,
/static/2e38d778d3d7f1a9c244a8bede600333/b6f2d/nessia.png 576w,
/static/2e38d778d3d7f1a9c244a8bede600333/17fa3/nessia.png 768w,
/static/2e38d778d3d7f1a9c244a8bede600333/6d185/nessia.png 2794w" sizes="(max-width: 384px) 100vw, 384px" /&gt;&lt;img loading="lazy" sizes="(max-width: 384px) 100vw, 384px" srcset="/static/2e38d778d3d7f1a9c244a8bede600333/32a58/nessia.png 96w,
/static/2e38d778d3d7f1a9c244a8bede600333/2c007/nessia.png 192w,
/static/2e38d778d3d7f1a9c244a8bede600333/75125/nessia.png 384w,
/static/2e38d778d3d7f1a9c244a8bede600333/b6f2d/nessia.png 576w,
/static/2e38d778d3d7f1a9c244a8bede600333/17fa3/nessia.png 768w,
/static/2e38d778d3d7f1a9c244a8bede600333/6d185/nessia.png 2794w" src="/static/2e38d778d3d7f1a9c244a8bede600333/75125/nessia.png" alt="NessIA homepage" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
      </noscript>
    </div>
    <div
      class="react-slidedown emotion-12"
      style="height: auto;"
    >
      <div
        class="emotion-10"
      >
        <h3
          class="emotion-3"
        >
          Business website
        </h3>
        <div
          class="emotion-8"
          data-testid="card-tag-container"
        >
          <span
            class="emotion-4"
          >
            javascript
          </span>
          <span
            class="emotion-5"
          >
            react
          </span>
          <span
            class="emotion-6"
          >
            gatsby
          </span>
          <span
            class="emotion-7"
          >
            bulma
          </span>
        </div>
        <p
          class="emotion-9"
        >
          Built for a client operating in the Business Intelligence space, this website outperforms competitors in speed, design and usability. It serves as a showcase for their services and as social proof via the Client and Partners section.
        </p>
      </div>
      <button
        class="emotion-11"
        data-testid="expand-button"
      >
        See More
      </button>
    </div>
  </div>
</div>
`;
